src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/Task.cs

```diff
@@ -5025,12 +5025,62 @@ private static int WaitAnyCore(Task[] tasks, int millisecondsTimeout, Cancellati
 
         #region FromResult / FromException / FromCanceled
 
-        /// <summary>Creates a <see cref="Task{TResult}"/> that's completed successfully with the specified result.</summary>
+        /// <summary>Gets a <see cref="Task{TResult}"/> that's completed successfully with the specified result.</summary>
         /// <typeparam name="TResult">The type of the result returned by the task.</typeparam>
         /// <param name="result">The result to store into the completed task.</param>
         /// <returns>The successfully completed task.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)] // method looks long, but for a given TResult it results in a relatively small amount of asm
         public static Task<TResult> FromResult<TResult>(TResult result)
```

Comment:
Most likely will easily hit inliner's time budget (if you try to use it in some already complicated method). E.g. https://github.com/dotnet/runtime/issues/43761#issuecomment-715538316

Reply:
> It's big and contains a lot of box/unbox so most likely will easily hit inliner's time budget (if you try to use it in some already complicated method)

It's used by async method's MoveNext methods, which are quite complicated, and it gets inlined there.